### PR TITLE
feat: display links to and descriptions of individual examples

### LINKED ISSUE #129: `source` link should point to specific example source, not the root of the Codex repo
`source` link should point to specific example source, not the root of the Codex repo.

### PR DESCRIPTION
**Pull request by definitelynobody**
Closes #129

image

### CODE REVIEW

File: src/examples.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    pub(crate) description: String,
+}
+
+impl fmt::Display for Example {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:#?}", self)
+    }
+}
+
+pub(crate) static EXAMPLES: Lazy<Vec<Example>> = Lazy::new(|| {
```
Comment (rvolosatovs) [inline src/examples.rs:22]: Just define an array of `Example` here directly, we don't gain anything by keeping the examples in toml and parsing at runtime, it only makes this more error-prone

**Review by rvolosatovs — CHANGES_REQUESTED** [review]

File: src/examples.rs (hunk 1 above)
Comment (rvolosatovs) [reply src/examples.rs:22]: Alternatively, even better, add this to the existing benefice config

Comment (definitelynobody) [reply src/examples.rs]: I added a --examples PATH arg so you can override the examples without a deployment. The default Examples.toml is verified with a unit test

File: Examples.toml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+[[examples]]
+slug = "examples/cryptle-rust:0.2.0"
+url = "https://github.com/enarx/cryptle/tree/main"
```
Comment (rvolosatovs) [inline Examples.toml:3]: These should be permalinks, https://github.com/enarx/cryptle/tree/v0.2.0

File: Examples.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+desc = "A secure multi-party Wordle clone."
+
+[[examples]]
+slug = "examples/echo-tcp-rust-mio:0.2.0"
+url = "https://github.com/enarx/codex/tree/main/Rust/mio-echo-tcp"
```
Comment (rvolosatovs) [inline Examples.toml:8]:
```suggestion
url = "https://github.com/enarx/codex/tree/v0.1.0/Rust/mio-echo-tcp"
```

File: src/main.rs [hunk 4]
```diff
@@ -190,6 +187,10 @@ struct Args {
     /// Whether to run the container in privileged mode.
     #[clap(long)]
     privileged: bool,
+
+    /// File to load examples from.
+    #[clap(long, default_value = "Examples.toml")]
```
Comment (rvolosatovs) [inline src/main.rs:192]: Rather than having a default to a path, use the `Examples.toml` form this repo when no path is set

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**